executors/blsKeysTimeCache.go

```diff
@@ -0,0 +1,67 @@
+package executors
+
+import "sync"
+
+type faultyBLSKeyInfo struct {
+	numSnoozeEvents uint32
+	addedTimestamp  int64
+}
+
+type blsKeysTimeCache struct {
+	maxSnoozeEvents     uint32
+	cacheExpiration     uint64
+	getCurrentTimestamp func() int64
+	mutFaultyBLSKeys    sync.Mutex
+	faultyBLSKeys       map[string]*faultyBLSKeyInfo
+}
+
+// ArgsBlsKeysTimeCache is the argument DTO used for the NewBLSKeysTimeCache constructor function
+type ArgsBlsKeysTimeCache struct {
+	MaxSnoozeEvents     uint32
+	CacheExpiration     uint64
+	GetCurrentTimestamp func() int64
+}
+
+// NewBLSKeysTimeCache creates a new instance of type BLS keys time cache
+func NewBLSKeysTimeCache(args ArgsBlsKeysTimeCache) (*blsKeysTimeCache, error) {
+	if args.GetCurrentTimestamp == nil {
+		return nil, errNilCurrentTimestampHandler
+	}
+
+	return &blsKeysTimeCache{
+		maxSnoozeEvents:     args.MaxSnoozeEvents,
+		cacheExpiration:     args.CacheExpiration,
+		getCurrentTimestamp: args.GetCurrentTimestamp,
+		faultyBLSKeys:       make(map[string]*faultyBLSKeyInfo),
+	}, nil
+}
+
+// ShouldNotify returns true if the current BLS key should trigger a notification event
+func (timeCache *blsKeysTimeCache) ShouldNotify(blsKey string) bool {
+	timeCache.mutFaultyBLSKeys.Lock()
+	defer timeCache.mutFaultyBLSKeys.Unlock()
+
+	currentTimestamp := timeCache.getCurrentTimestamp()
+
+	info := timeCache.faultyBLSKeys[blsKey]
+	if info == nil {
+		info = &faultyBLSKeyInfo{
+			addedTimestamp: currentTimestamp,
+		}
+		timeCache.faultyBLSKeys[blsKey] = info
```

Comment:
This in-memory storage is not cleaned / entries are not removed, but that is not a concern (as there are only a few items, at most thousands) :+1:

Reply:
exactly. Didn't want it to complicate it with cleaning go routines. Can be added at a later time, if desired. :)